### PR TITLE
Sets up NFS client config on 4 nodes

### PR DESCRIPTION
**Pull request by navidshaikh**
This is to configure 4 nodes of CCCP deployment in CI as NFS clients.
  NFS server configurations can be configured via group variable.

### CODE REVIEW

**Comment by navidshaikh** [comment]
#dotests

**Comment by navidshaikh** [comment]
#dotests

File: provisions/group_vars/all [hunk 1]
```diff
@@ -9,3 +9,5 @@ cccp_source_branch: master
 public_registry: registry.centos.org
 
 enable_epel: true
+
+test_nfs_share: 172.29.32.44:/nfsshare
```
Comment (navidshaikh) [inline provisions/group_vars/all]: This is NFS share created in devcloud serving the range of clients in devcloud. ( @bamachrn created it)

**Comment by navidshaikh** [comment]
#dotests

**Comment by dharmit** [comment]
This is what I've been doing for NFS mounts on my DevCloud setup:

```bash
$ onevm list | awk '{print $1}' # this prints only VM IDs
ID
611
894
895
896
897

$ for i in `seq 894 897`; do myssh $i '( mkdir /srv/pipeline-logs )'; done
$ for i in `seq 894 897`; do myssh $i '( mount -t nfs 172.29.32.44:/nfsshare /srv/pipeline-logs )'; done
```

So, as far as I can see, we need to do only two things on the slaves in DevCloud:
- create a directory
- mount the NFS share

Installation of packages can be skipped AFAICT.

File: provisions/group_vars/all (hunk 1 above)
Comment (rtnpro) [inline provisions/group_vars/all]: @navidshaikh CI nodes cannot access devcloud. So, we need to setup another CI node, if it's needed.

Comment (navidshaikh) [inline provisions/group_vars/all]: @rtnpro ACK!

**Comment by navidshaikh** [comment]
> Installation of packages can be skipped AFAICT.
@dharmit : done.

**Comment by rtnpro** [comment]
It seems like you are having an error in Ansible syntax:

```
ERROR! 'when' is not a valid attribute for a Play

The error appears to have been in '/root/container-pipeline-service/provisions/vagrant.yml': line 15, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Configure NFS server on scanner node for CI
  ^ here

Build failed: Pseudo-terminal will not be allocated because stdin is not a terminal.
Warning: Permanently added 'n64.dusty.ci.centos.org,172.19.2.128' (ECDSA) to the list of known hosts.
 [WARNING]: Optional dependency 'cryptography' raised an exception, falling
back to 'Crypto'
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and 
make sure become_method is 'sudo' (default).
This feature will be removed in a 
future release. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
ERROR! 'when' is not a valid attribute for a Play

The error appears to have been in '/root/container-pipeline-service/provisions/vagrant.yml': line 15, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Configure NFS server on scanner node for CI
  ^ here
```

**Comment by navidshaikh** [comment]
> It seems like you are having an error in Ansible syntax:

```
ERROR! 'when' is not a valid attribute for a Play
```

Yes, updated the PR. I was using `when` in a play.

**Comment by navidshaikh** [comment]
#dotests

**Comment by navidshaikh** [comment]
#dotests

**Comment by navidshaikh** [comment]
#dotests

**Comment by bamachrn** [comment]
How are we mounting the NFS mount points for CI and dev? I know we did this by adding entry in fstab for prod. we need the mount points in all the nodes roght?
@navidshaikh

**Comment by navidshaikh** [comment]
> How are we mounting the NFS mount points for CI and dev? I know we did this by adding entry in fstab for prod. 

The PR should how we are configuring the NFS mounts in CI.  For dev env, we can configure the NFS mounts with `test=True`.

> we need the mount points in all the nodes roght?
Yes!

**Comment by bamachrn** [comment]
LGTM

**Pull request merged**